### PR TITLE
Update Sundials common interface

### PR DESCRIPTION
**Pull request by YingboMa**

### CODE REVIEW

File: src/API/arkode.jl [hunk 1]
```diff
@@ -66,16 +66,16 @@ function ARKStepResFtolerance(arkode_mem, rfun::ARKRwtFn)
     ccall((:ARKStepResFtolerance, libsundials_arkode), Cint, (ARKStepMemPtr, ARKRwtFn), arkode_mem, rfun)
 end
 
-function ARKStepSetLinearSolver(arkode_mem, LS::SUNLinearSolver, A::SUNMatrix)
+function ARKStepSetLinearSolver(arkode_mem, LS::SUNLinearSolver, A)
```
Comment (ChrisRackauckas) [inline src/API/arkode.jl:69]: it's always dangerous directly modifying the wrapped API files. Instead, make an API additions file for these dispatches.

Comment (jd-lara) [reply src/API/arkode.jl:69]: We should probably do the same here as with NVector and N_Vector probably and have a conversion method and add another function that dispatches on the type.

Comment (jd-lara) [reply src/API/arkode.jl:69]: We can add some conversion method for these function and have a similar behavior as N_Vector and NVector also.

**Comment by YingboMa** [comment]
Mac CIs seem to have stuck, so I will merge this PR.

**Comment by ChrisRackauckas** [comment]
Mac CI is terrible. We can probably turn those off once we're comfortable the binaries work well.

**Pull request merged**